File: flash/models/page_activity.go
package models

import "time"

type PageActivity struct {
	ID        uint   `gorm:"primaryKey" json:"id"`
	ProjectID uint64 `json:"project_id"`
	Type      string `json:"type"`
	PageURL   string `json:"page_url"`
	IPAddress string `json:"ip_address"`

	ModelID   *uint64 `json:"model_id"`
	ModelName *string `json:"model_name"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: apps/web-admin/app/Models/PageActivity.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class PageActivity extends Model
{
    use HasFactory;

    protected $fillable = [
        'project_id',
        'type',
        'page_url',
        'ip_address',
        'model_id',
        'model_name',
    ];

    public function project(): BelongsTo
    {
        return $this->belongsTo(Project::class);
    }
}
